pkg/helmreconciler/dependencies.go

```diff
@@ -0,0 +1,31 @@
+package helmreconciler
+
+import (
+	"io"
+	"strings"
+
+	"istio.io/operator/pkg/name"
+)
+
+func InsertChildrenRecursive(componentName name.ComponentName, tree ComponentTree, children ComponentNameToListMap) {
+	tree[componentName] = make(ComponentTree)
+	for _, child := range children[componentName] {
+		InsertChildrenRecursive(child, tree[componentName].(ComponentTree), children)
+	}
+}
+
+func InstallTreeString(ct ComponentTree) string {
+	var sb strings.Builder
+	buildInstallTreeString(ct, name.IstioBaseComponentName, "", &sb)
+	return sb.String()
+}
+
+func buildInstallTreeString(ct ComponentTree, componentName name.ComponentName, prefix string, sb io.StringWriter) {
+	_, _ = sb.WriteString(prefix + string(componentName) + "\n")
+	if _, ok := ct[componentName].(ComponentTree); !ok {
+		return
+	}
+	for k := range ct[componentName].(ComponentTree) {
+		buildInstallTreeString(ct, k, prefix+"  ", sb)
```

Comment:
this is fine for this PR, but I really really think you, or I, or someone else working on this code should reimplement this recursion into a non-recursive implementation.  recursion is hard to test and hard to follow.